src/entries/popup/pages/messages/useHasEnoughGas.ts

```diff
@@ -14,10 +16,11 @@ export const useHasEnoughGas = (session: ActiveSession) => {
     chainId,
   });
   const selectedGas = useGasStore.use.selectedGas();
-
-  console.log('nativeAsset', chainId, nativeAsset);
-  return lessThan(
-    selectedGas?.gasFee?.amount || '0',
-    toWei(nativeAsset?.balance?.amount || '0'),
-  );
+  const hasEnough = useMemo(() => {
+    return lessThan(
+      selectedGas?.gasFee?.amount || '0',
+      toWei(nativeAsset?.balance?.amount || '0'),
+    );
+  }, [selectedGas?.gasFee?.amount, nativeAsset?.balance?.amount]);
+  return hasEnough;
```

Comment:
I saw some pretty intense rerendering going on within our dapp request components: https://www.loom.com/share/f8c651f7282f4beeac3ed9fd5a8652b5

This lessens the rerenders slightly and improved behavior a bit during testing but I think the source of the issue stems from some pretty aggressive gas polling worsened by some inefficient rendering.

Wondering if we should throw this vid into a ticket and revisit `useGas`. Maybe we need to play with debounce or throw `enabled` flags on gas hooks and only enable polling in certain UI states.

Reply:
+1 on creating a ticket with high priority in the TO DO status